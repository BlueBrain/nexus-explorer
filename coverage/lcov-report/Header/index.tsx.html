<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Header/index.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">Header</a> index.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">71.43% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/9</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import * as React from 'react';
import { Menu, Dropdown, Icon } from 'antd';
import './Header.less';
&nbsp;
const logo = require('../../logo.svg');
&nbsp;
export interface HeaderProps {
  name?: string;
  links?: React.ReactNode[];
  displayLogin?: boolean;
  children?: React.ReactChild;
  onLoginClick?(): void;
}
&nbsp;
const Header: React.FunctionComponent&lt;HeaderProps&gt; = ({
  name = <span class="branch-0 cbranch-no" title="branch not covered" >'',</span>
  displayLogin = true,
  links = <span class="branch-0 cbranch-no" title="branch not covered" >[],</span>
  children,
  onLoginClick,
}) =&gt; {
  const menu = (
    &lt;Menu&gt;
      {links.map((link, i) =&gt; (
        &lt;Menu.Item key={i}&gt;{link}&lt;/Menu.Item&gt;
      ))}
    &lt;/Menu&gt;
  );
&nbsp;
  return (
    &lt;header className="Header"&gt;
      &lt;div className="logo-block"&gt;
        &lt;a className="logo" href=""&gt;
          {/* inline styling to prevent headaches in dev mode */}
          &lt;img src={logo} alt="Nexus" style={{ height: '2em', width: '2em' }} /&gt;
        &lt;/a&gt;
        &lt;h1&gt;Nexus&lt;/h1&gt;
      &lt;/div&gt;
      &lt;div className="selectors"&gt;{children}&lt;/div&gt;
      &lt;div className="menu-block"&gt;
        {name ? (
          &lt;Dropdown overlay={menu}&gt;
            &lt;a className="menu-dropdown ant-dropdown-link"&gt;
              {name} &lt;Icon type="down" /&gt;
            &lt;/a&gt;
          &lt;/Dropdown&gt;
        ) : displayLogin ? (
          &lt;a className="menu-dropdown ant-dropdown-link" onClick={onLoginClick}&gt;
            login &lt;Icon type="login" /&gt;
          &lt;/a&gt;
        ) : null}
      &lt;/div&gt;
    &lt;/header&gt;
  );
};
&nbsp;
export default Header;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jan 16 2019 08:49:36 GMT+0100 (CET)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
