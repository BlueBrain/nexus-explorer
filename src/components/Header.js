import React from "react";
import logo from "../../public/img/logo.png";
import SearchBar from './SearchBar';
import Login from './Login';
import { WithStore } from '@bbp/nexus-react';
import { navigate } from '../store/actions';

const Header = base => (
  <header>
    <div className="logo-block">
      <a className="logo" href={base + "/"}>
        <img src={logo} />
      </a>
      <WithStore
        mapStateToProps={(state) => ({
          config: state.config
        })}
        mapDispatchToProps={{
          navigate: navigate.navigate
        }}
      >{({ config, navigate }) => <a href={config.appPath ? config.appPath : "/" } onClick={() => navigate('')}><h1>Explorer</h1></a>}
      </WithStore>
    </div>
    <SearchBar />
    <Login />
  </header>
);

export default Header;
