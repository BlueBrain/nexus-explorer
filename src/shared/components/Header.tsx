import React = require('react');
import { Link } from 'react-router-dom';

export interface LinkProp {
  name: string;
  url: string;
}

export interface HeaderProps {
  links: LinkProp[];
}

const Header: React.StatelessComponent<HeaderProps> = ({ links }) => (
  <ul>
    {links.map(link => <li key={link.url}>
      <Link
        to={link.url}
      >{link.name}</Link>
    </li>)}
  </ul>
);

export default Header;
