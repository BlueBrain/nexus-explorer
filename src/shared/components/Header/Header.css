.Header {
  background-color: #ff6666;
}