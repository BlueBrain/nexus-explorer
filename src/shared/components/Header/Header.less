@import '../../lib.less';

.Header {
  display: flex;
  justify-content: space-between;
  box-sizing: content-box;
  height: 52px;
  background-color: #ffffff;
  box-shadow: 2px 3px 3px @subtle-white;

  .selectors {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    flex-grow: 1;
    margin-left: 10px;

    .OrgDropdown {
      height: 100%;
    }
  }

  .menu-block {
    display: flex;
    justify-content: flex-end;
    flex-grow: 1;
    padding-right: @default-pad;
    align-items: center;

    button,
    a {
      margin: 5px;
      font-weight: lighter;
    }

    .menu-dropdown {
      color: @text-color;
      align-self: center;
      margin-left: 10px;
      bottom: 50px;
    }
  }

  .ui-header-info-button {
    margin: 5px 5px 4px;
    cursor: pointer;
  }

  .copy-token-button {
    height: 30px;
    width: 115px;
    border-radius: 4px;
    outline-style: none;
    line-height: 1rem;
    cursor: pointer;
    font-size: 0.9rem;
    text-align: left;
    border: 1px solid @subtle-white;
    display: flex;
    align-items: center;
    background-color: inherit;

    .button-text {
      margin-left: 0.4rem;
    }
  }
}

.nav-item {
  font-size: 16px;
}

.popover-title {
  padding: 0 0 5px 0;
}

// the gtm modal should appear on a top of a popover
.ant-popover {
  z-index: 1000 !important;
}

/*
  ##Device = Tablets, Ipads (portrait)
  ##Screen = B/w 768px to 1024px
*/

@media (max-width: 1024px) {
  .hide-tablet {
    display: none;
  }
}

.menu-icon {
  border: none;
  outline-style: none;

  img {
    width: 17px;
    margin-left: 1rem;
  }

  img > svg {
    fill: white;
  }
}
