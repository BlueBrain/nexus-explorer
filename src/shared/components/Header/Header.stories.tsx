import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Header from './index';

storiesOf('Components/Header', module).add(
  'Header',
  withInfo(`
    The header component is used throughout the application.

    It is used to display the user name and navigation links

    ~~~js
    <Header name="Mark Hamill" links={[
      <a href="#">Home</a>,
      <a href="#">Settings</a>,
    ]} />
    ~~~
  `)(() => (
    <div style={{ margin: '50px 40px 0px' }}>
      <Header
        name="Mark Hamill"
        links={[<a href="#">Home</a>, <a href="#">Settings</a>]}
      />
    </div>
  ))
);
