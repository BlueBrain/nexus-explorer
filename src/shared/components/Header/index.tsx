import React = require('react');
import './Header.less';

export interface HeaderProps {
  links: React.ReactNode[];
}

const Header: React.StatelessComponent<HeaderProps> = ({ links }) => (
  <div className="Header">
    <ul>
      {links.map((link, i) => <li key={i}>{link}</li>)}
    </ul>
  </div>
);

export default Header;
