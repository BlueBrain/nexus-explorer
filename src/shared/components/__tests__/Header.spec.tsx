import React = require('react');
import { shallow, render } from 'enzyme';
import Header, { LinkProp } from '../Header';

const links: LinkProp[] = [
  {
    name: 'lol',
    url: '/lol',
  },
];

describe('Header component', () => {
  it('Should render correctly', () => {
    const tree = shallow(<Header links={links} />, { context: {} });
    expect(tree).toMatchSnapshot();
  });
});
