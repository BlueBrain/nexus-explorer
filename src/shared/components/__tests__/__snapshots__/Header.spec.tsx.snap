// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render correctly 1`] = `
<ul>
  <li
    key="/lol"
  >
    <Link
      replace={false}
      to="/lol"
    >
      lol
    </Link>
  </li>
</ul>
`;
